### PR TITLE
Improve Batch documentation

### PR DESCRIPTION
**Pull request by matthchr**
- Improved application package references documentation.
  - Improved max tasks per node documentation.
  - Improved spread vs pack documentation.
  - Improved documentation about task idempotency.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by matthchr** [comment]
@bgklein, @darylmsft, @jafreck please feel free to review this updated text

**Please hold off merging until I give the go-ahead after the text has been reviewed**

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json [hunk 1]
```diff
@@ -11836,11 +11836,12 @@
           "type": "integer",
           "format": "int32",
           "title": "The maximum number of tasks that can run concurrently on a single compute node in the pool.",
-          "description": "The default value is 1. The maximum value of this setting depends on the size of the compute nodes in the pool (the vmSize setting)."
+          "description": "The default value is 1. The maximum value is 4 times the number of cores of the vmSize of the pool, or 256, whichever is smaller."
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]:
```suggestion
          "description": "The default value is 1. The maximum value is the smaller of 4 times the number of cores of the vmSize of the pool or 256"
```
I don't really like the two commas. Either this or get rid of the comma in 'vmSize of the pool or 256'

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json [hunk 2]
```diff
@@ -13367,7 +13369,8 @@
           "items": {
             "$ref": "#/definitions/ApplicationPackageReference"
           },
-          "title": "The list of application packages to be installed on each compute node in the pool."
+          "title": "The list of application packages to be installed on each compute node in the pool.",
+          "description": "Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There can be a maximum of 10 application package references on any given pool."
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]:
```suggestion
          "description": "Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There is a maximum of 10 application package references on any given pool."
```

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json [hunk 3]
```diff
@@ -11902,7 +11903,8 @@
...
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]:
```suggestion
          "description": "Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There can is a maximum of 10 application package references on any given pool."
```

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json [hunk 4]
```diff
@@ -13380,11 +13383,13 @@
         "maxTasksPerNode": {
           "type": "integer",
           "format": "int32",
-          "title": "The maximum number of tasks that can run concurrently on a single compute node in the pool."
+          "title": "The maximum number of tasks that can run concurrently on a single compute node in the pool.",
+          "description": "The default value is 1. The maximum value is 4 times the number of cores of the vmSize of the pool, or 256, whichever is smaller."
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]:
```suggestion
          "description": "The default value is 1. The maximum value is the smaller of 4 times the number of cores of the vmSize of the pool or 256"
```
I don't really like the two commas. Either this or get rid of the comma in 'vmSize of the pool or 256'

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json [hunk 5]
```diff
@@ -13510,11 +13516,12 @@
           "type": "integer",
           "format": "int32",
           "title": "The maximum number of tasks that can run concurrently on a single compute node in the pool.",
-          "description": "The default value is 1. The maximum value of this setting depends on the size of the compute nodes in the pool (the vmSize setting)."
+          "description": "The default value is 1. The maximum value is 4 times the number of cores of the vmSize of the pool, or 256, whichever is smaller."
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]:
```suggestion
          "description": "The default value is 1. The maximum value is the smaller of 4 times the number of cores of the vmSize of the pool or 256"
```
I don't really like the two commas. Either this or get rid of the comma in 'vmSize of the pool or 256'

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json [hunk 6]
```diff
@@ -14982,8 +14989,8 @@
...
             "$ref": "#/definitions/ApplicationPackageReference"
           },
-          "title": "A list of application packages to be installed on each compute node in the pool.",
-          "description": "The list replaces any existing application package references on the pool. Changes to application package references affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. If omitted, or if you specify an empty collection, any existing application packages references are removed from the pool."
+          "title": "The list of application packages to be installed on each compute node in the pool.",
+          "description": "The list replaces any existing application package references on the pool. Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There can be a maximum of 10 application package references on any given pool. If omitted, or if you specify an empty collection, any existing application packages references are removed from the pool."
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]:
```suggestion
          "description": "The list replaces any existing application package references on the pool. Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There is a maximum of 10 application package references on any given pool. If omitted, or if you specify an empty collection, any existing application packages references are removed from the pool."
```

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json [hunk 7]
```diff
@@ -13496,7 +13501,8 @@
           "items": {
             "$ref": "#/definitions/ApplicationPackageReference"
           },
-          "title": "The list of application packages to be installed on each compute node in the pool."
+          "title": "The list of application packages to be installed on each compute node in the pool.",
+          "description": "Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There can be a maximum of 10 application package references on any given pool."
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]:
```suggestion
          "description": "Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There is a maximum of 10 application package references on any given pool."
```

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json [hunk 8]
```diff
@@ -15017,8 +15024,8 @@
...
           },
-          "title": "A list of application packages to be installed on each compute node in the pool.",
-          "description": "Changes to application package references affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. If this element is present, it replaces any existing application package references. If you specify an empty collection, then all application package references are removed from the pool. If omitted, any existing application package references are left unchanged."
+          "title": "The list of application packages to be installed on each compute node in the pool.",
+          "description": "The list replaces any existing application package references on the pool. Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There can be a maximum of 10 application package references on any given pool. If omitted, any existing application package references are left unchanged."
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]:
```suggestion
          "description": "The list replaces any existing application package references on the pool. Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There is a maximum of 10 application package references on any given pool. If omitted, any existing application package references are left unchanged."
```

Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]: If you are changing 'Changes to application package references' => 'Changes to application packages' also apply to 'The list replaces any existing application package references on the pool.' and other references in the body of text

File: specification/batch/resource-manager/Microsoft.Batch/stable/2019-04-01/BatchManagement.json [hunk 9]
```diff
@@ -2772,11 +2772,13 @@
         "maxTasksPerNode": {
           "type": "integer",
           "format": "int32",
-          "title": "The maximum number of tasks that can run concurrently on a single compute node in the pool."
+          "title": "The maximum number of tasks that can run concurrently on a single compute node in the pool.",
+          "description": "The default value is 1. The maximum value is 4 times the number of cores of the vmSize of the pool, or 256, whichever is smaller."
```
Comment (bgklein) [inline specification/batch/resource-manager/Microsoft.Batch/stable/2019-04-01/BatchManagement.json]:
```suggestion
          "description": "The default value is 1. The maximum value is the smaller of 4 times the number of cores of the vmSize of the pool or 256"
```
I don't really like the two commas. Either this or get rid of the comma in 'vmSize of the pool or 256'

File: specification/batch/resource-manager/Microsoft.Batch/stable/2019-04-01/BatchManagement.json [hunk 10]
```diff
@@ -2812,7 +2814,7 @@
             "$ref": "#/definitions/ApplicationPackageReference"
           },
           "title": "The list of application packages to be installed on each compute node in the pool.",
-          "description": "Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged."
+          "description": "Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There can be a maximum of 10 application package references on any given pool."
```
Comment (bgklein) [inline specification/batch/resource-manager/Microsoft.Batch/stable/2019-04-01/BatchManagement.json]:
```suggestion
          "description": "Changes to application packages affect all new compute nodes joining the pool, but do not affect compute nodes that are already in the pool until they are rebooted or reimaged. There is a maximum of 10 application package references on any given pool."
```

**Review by bgklein — CHANGES_REQUESTED** [review]

**Comment by matthchr** [comment]
Some of the checks seem to be failing due to: `Failed to load error: libunwind.so.8: cannot open shared object file: No such file or directory`

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json [hunk 11]
```diff
@@ -11836,11 +11836,12 @@
           "type": "integer",
           "format": "int32",
           "title": "The maximum number of tasks that can run concurrently on a single compute node in the pool.",
-          "description": "The default value is 1. The maximum value of this setting depends on the size of the compute nodes in the pool (the vmSize setting)."
+          "description": "The default value is 1. The maximum value is the smaller of 4 times the number of cores of the vmSize of the pool or 256."
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]: Just realized how many 'of x' are in this sentence any way to simplify?

**Review by sarangan12 — APPROVED** [review]

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json (hunk 11 above)
Comment (matthchr) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json]: Yeah I realized it when writing it too but not sure how to improve: "4 * cores of the vm size of the pool" which is the main culprit

**Review by bgklein — APPROVED** [review]

**Comment by matthchr** [comment]
@sarangan12 Feel free to merge this now, as one of my coworkers has approved.

I see quite a few CI steps failing but as far as I can tell they are all due to CI failures (see my comment about libunwind.so.8 above) and not my changes...

**Review by sarangan12 — APPROVED** [review]

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**